### PR TITLE
Minor cleanup of model developer documentation

### PR DESCRIPTION
**Pull request by wouterdb**
# Description

I did this cleanup in preparation for some other docs changes 

1. fixed some dead links
2. renamed a few sections to be clearer
3. fixed order in the menu to logical order (instead of alphabetical)
4. added coockiecutter to project guide

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] ~~Attached issue to pull request~~
- [X] ~~Changelog entry~~
- [ ] ~~Type annotations are present~~
- [ ] ~~Code is clear and sufficiently documented~~
- [ ] ~~No (preventable) type errors (check using make mypy or make mypy-diff)~~
- [ ] ~~Sufficient test cases (reproduces the bug/tests the requested feature)~~
- [x] Correct, in line with design
- [x] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

# Reviewer Checklist:

- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented
- [ ] Correct, in line with design

### CODE REVIEW

**Review by bartv — COMMENTED** [review]
Is the removed information available in other parts of the documentation?

**Comment by wouterdb** [comment]
@bartv in detail, in the file configurationmodel.rst

- the note up top: no (or at least, I don't know, but is was very hard to understand in the flow anyways)
- options in project.yml, yes and link to complete list is below the shortened list
- deploy a file example, no, I can link to quickstart if you think that is relevant

**Comment by bartv** [comment]
> @bartv in detail, in the file configurationmodel.rst
> 
> * the note up top: no (or at least, I don't know, but is was very hard to understand in the flow anyways)
> * options in project.yml, yes and link to complete list is below the shortened list
> * deploy a file example, no, I can link to quickstart if you think that is relevant

No, I was mainly thinking about the git repo and format of requires

**Comment by wouterdb** [comment]
git is initialised by cookiecutter, but if there are thing I should beter put back, it would be easiest if you add a comment to that particular change, as I don't know exactly what you are referring to.

File: docs/model_developers/configurationmodel.rst [hunk 1]
```diff
@@ -12,73 +13,62 @@ contains the source code of the configuration model. This project also matches w
 :term:`project` defined on the server, from which multiple :term:`environments<environment>` can be
 deployed.
 
-.. note::
-
-    Inmanta requires that a project is a git repository. This is not strictly required when using
```
Comment (bartv) [inline docs/model_developers/configurationmodel.rst:17]: Is this information still present somewhere in the documentation?

File: docs/model_developers/configurationmodel.rst [hunk 2]
```diff
@@ -12,73 +13,62 @@ contains the source code of the configuration model. This project also matches w
...
+    * ``author``  The author of the module
+    * ``author_email`` The contact email address of author
+    * ``license`` License the module is released under
+    * ``copyright`` Copyright holder name and date.
     * ``install_mode`` This key determines what version of a module should be selected when a module
       is downloaded. This is used when the module version is not "pinned" in the ``requires`` list.
       The available values are:
 
         * release (default): Only use a released version, that is compatible with the current
-          compiler. A version is released when there is a tag on a commit. This tag should be a
```
Comment (bartv) [inline docs/model_developers/configurationmodel.rst:47]: Is this explained somewhere else in the documentation?

File: docs/reference/projectyml.rst [hunk 3]
```diff
@@ -2,18 +2,24 @@ Compiler Configuration Reference
...
 modules. 
 
-For basic usage information, see 'Create a configuration model'_.
+For basic usage information, see :ref:`project-creation-guide`.
 
 Project.yml defines the following settings:
 
     * ``name`` An optional name of the project.
     * ``description`` An optional description of the project
+    * ``author``  The author of the module
```
Comment (arnaudsjs) [inline docs/reference/projectyml.rst:19]: Also change `An optional name of the project.` -> `The name of the project.`

**Review by arnaudsjs — CHANGES_REQUESTED** [review]

File: docs/model_developers/configurationmodel.rst (hunk 1 above)
Comment (wouterdb) [reply docs/model_developers/configurationmodel.rst:17]: added part of the note back

File: docs/model_developers/configurationmodel.rst (hunk 2 above)
Comment (wouterdb) [reply docs/model_developers/configurationmodel.rst:47]: yes, linked below: :ref:`project_yml`

File: docs/reference/projectyml.rst (hunk 3 above)
Comment (wouterdb) [reply docs/reference/projectyml.rst:19]: done

**Review by arnaudsjs — APPROVED** [review]

**Review by bartv — APPROVED** [review]

**Pull request merged**